app/src/main/java/com/toshi/manager/chat/tasks/BaseGroupTask.kt

```
@@ -0,0 +1,31 @@
+/*
+ * 	Copyright (c) 2017. Toshi Inc
+ *
+ * 	This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.toshi.manager.chat.tasks
+
+import org.whispersystems.signalservice.api.push.exceptions.EncapsulatedExceptions
+import rx.Completable
+
+abstract class BaseGroupTask {
+    // Ignore EncapsulatedExceptions if an user has signed out
+    protected fun handleException(throwable: Throwable): Completable {
+        if (throwable is EncapsulatedExceptions && !throwable.unregisteredUserExceptions.isEmpty()) {
```

You can not just ignore `EncapsulatedExceptions`. You're getting an exception for a reason -- something exceptional is happening. Ignoring exceptions isn't a solution - it's just ignoring the problem.

What happens is a user has signed out; does this change ensure all signed in users in a group with the signed out user are getting their updates. Is the UI showing that one of the users is unreachable?

---

I'm only ignoring the exception when `unregisteredUserExceptions` isn't empty.

The UI shows a toast if an error occurs. 

So how to you want to handle this?

---

That didn't really answer my questions.

If the signed out user is in a group -- are all other signed in members of the group getting correctly updated?
Do we (should we) show an UI when a signed out user is in a group? I know that we show an error message if it is an individual conversation; how should we handle the group case.

---

Yes, all other signed in members of the group is updated correctly and throws an error.
I'll ask design

---

FYI, this is still super dangerous code. If `EncapsulatedExceptions` contains other errors alongside "unregistered user exceptions" your code will ignore them as well and return `Completable.complete`. You should really think about handling exceptions other than "unregistered user" as well.